pkg/api/api.go

```diff
@@ -263,6 +303,43 @@ func (s *server) newTracingHandler(spanName string) func(h http.Handler) http.Ha
 	}
 }
 
+func (s *server) permissionCheckHandler() func(h http.Handler) http.Handler {
```

Comment:
This middleware should not return plain text responses as it is part of the json-encoded endpoints. I think that there should be consistency, as clients may assume json encoded response even without checking content-type.